File: frontend/api/bookings.py
from .client import client


def get_event_seats(event_id, token):

    response = client.get(
        f"/events/{event_id}/seats/",
        token,
    )

    if response.status_code == 200:
        return response.json()

    return None


def hold_seat(event_id, seat_id, token):

    response = client.post(
        "/bookings/hold/",
        {
            "event": event_id,
            "seat": seat_id,
        },
        token,
    )

    return response


def create_booking(
    event_id,
    seat_id,
    token,
):

    response = client.post(
        "/bookings/",
        {
            "event": event_id,
            "seat": seat_id,
        },
        token,
    )

    return response

def get_my_bookings(token):
    response = client.get(
        "/bookings/my-bookings/",
        token,
    )

    if response.status_code == 200:
        return response.json()

    return []

def cancel_booking(booking_id, token):

    return client.patch(
        f"/bookings/{booking_id}/cancel/",
        token,
    )

File: frontend/src/api/bookings.js
import apiClient from "./client";

// ============================================================
// GET EVENT SEATS
// ============================================================

export const getEventSeats = async (eventId) => {
  const response = await apiClient.get(
    "/bookings/seats/",
    {
      params: {
        event: eventId,
      },
    }
  );

  return response.data;
};

// ============================================================
// HOLD SEAT
// ============================================================

export const holdSeat = async (
  eventId,
  seatId
) => {
  const response = await apiClient.post(
    "/bookings/hold/",
    {
      event: eventId,
      seat: seatId,
    }
  );

  return response.data;
};

// ============================================================
// CHECK HOLD STATUS
// ============================================================

export const getSeatHoldStatus = async (
  eventId,
  seatId
) => {
  const response = await apiClient.get(
    "/bookings/hold-status/",
    {
      params: {
        event: eventId,
        seat: seatId,
      },
    }
  );

  return response.data;
};

// ============================================================
// CREATE BOOKING
// ============================================================

export const createBooking = async (
  eventId,
  seatId
) => {

  const response = await apiClient.post(
    "/bookings/",
    {
      event: eventId,
      seat: seatId,
    }
  );

  return response.data;
};

// ============================================================
// GET MY BOOKINGS
// ============================================================

export const getMyBookings = async () => {

  const response = await apiClient.get(
    "/bookings/my-bookings/"
  );

  return response.data;
};

// ============================================================
// CANCEL BOOKING
// ============================================================

export const cancelBooking = async (
  bookingId
) => {

  const response = await apiClient.patch(
    `/bookings/${bookingId}/cancel/`
  );

  return response.data;
};